### ISSUE TITLE
Download unavailable after click on Download in New Repository

### ISSUE THREAD

**Issue by peter-ogden** [hacs/integration#2291]
### Web browser

Chrome

### Web browser version

95.0.4638.69

## System Health

version | core-2021.11.3
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.9.7
os_name | Linux
os_version | 5.10.75
arch | x86_64
timezone | Australia/Perth

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
Github API Calls Remaining | 4882
Installed Version | 1.17.2
Stage | running
Available Repositories | 909
Installed Repositories | 9

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | true
-- | --
subscription_expiration | November 18, 2021, 8:00 AM
relayer_connected | true
remote_enabled | true
remote_connected | true
alexa_enabled | false
google_enabled | true
remote_server | ap-southeast-1-0.ui.nabu.casa
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 6.6
-- | --
update_channel | stable
supervisor_version | supervisor-2021.10.8
docker_version | 20.10.8
disk_total | 30.8 GB
disk_used | 9.6 GB
healthy | true
supported | true
board | ova
supervisor_api | ok
version_api | ok
installed_addons | Duck DNS (1.14.0), Samba share (9.5.1), NGINX Home Assistant SSL proxy (3.0.2), Node-RED (10.1.0), Mosquitto broker (6.0.1), AdGuard Home (4.2.0), MariaDB (2.4.0), ESPHome (2021.10.3), Home Assistant Google Drive Backup (0.105.2)

</details>

<details><summary>Lovelace</summary>

dashboards | 2
-- | --
resources | 4
views | 6
mode | storage

</details>

### Checklist

- [X] I'm running the newest version of HACS <https://github.com/hacs/integration/releases/latest>
- [X] I have filled out the issue template to the best of my ability.
- [X] I have read <https://hacs.xyz/docs/issues>
- [X] This issue is related to the frontend of HACS.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).

### Describe the issue

Updated HACS to 1.17.2 and restarted Home Assistant. New repositories were then displayed as being available. I browsed through the list of new repositories and dismissed them as I was not interested in most, but there was one Lovelace Card available that I wanted to download and try (Bom Radar Card), but when I clicked on Download, the Download button on the dialog box was greyed out and I was unable to proceed.

I understand that this could actually be an issue with the repository, but there is no error message in the UI nor other information to indicate this.

Core Log:
[2021-11-15 10:37:37 WARNING (SyncWorker_2) [homeassistant.loader] We found a custom integration bureau_of_meteorology which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2021-11-15 10:37:37 WARNING (SyncWorker_1) [homeassistant.loader] We found a custom integration nodered which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2021-11-15 10:37:37 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration sems which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2021-11-15 10:37:37 WARNING (SyncWorker_1) [homeassistant.loader] We found a custom integration hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2021-11-15 10:37:37 WARNING (SyncWorker_3) [homeassistant.loader] We found a custom integration average which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2021-11-15 10:38:09 ERROR (MainThread) [snitun.client.client_peer] Challenge/Response error with SniTun server (0 bytes read on a total of 32 expected bytes)
2021-11-15 10:38:09 ERROR (MainThread) [hass_nabucasa.remote] Connection problem to snitun server
2021-11-15 10:38:14 ERROR (MainThread) [homeassistant] Error doing job: Task was destroyed but it is pending!
2021-11-15 11:05:05 ERROR (MainThread) [custom_components.hacs] Could not use repository with ID 0 ('NoneType' object has no attribute 'data')
2021-11-15 11:05:37 ERROR (MainThread) [custom_components.hacs] Could not use repository with ID 0 ('NoneType' object has no attribute 'data')
2021-11-15 11:12:55 ERROR (MainThread) [custom_components.hacs] Could not use repository with ID 0 ('NoneType' object has no attribute 'data')
2021-11-15 11:28:33 ERROR (MainThread) [custom_components.hacs] Could not use repository with ID 0 ('NoneType' object has no attribute 'data')
2021-11-15 11:29:33 ERROR (MainThread) [custom_components.hacs] Could not use repository with ID 0 ('NoneType' object has no attribute 'data')
2021-11-15 11:32:41 ERROR (MainThread) [custom_components.hacs] Could not use repository with ID 0 ('NoneType' object has no attribute 'data')
2021-11-15 11:32:57 ERROR (MainThread) [custom_components.hacs] Could not use repository with ID 0 ('NoneType' object has no attribute 'data')
2021-11-15 11:34:06 ERROR (MainThread) [custom_components.hacs] Could not use repository with ID 0 ('NoneType' object has no attribute 'data')
2021-11-15 11:35:45 ERROR (MainThread) [custom_components.hacs] Could not use repository with ID 0 ('NoneType' object has no attribute 'data')

### Reproduction steps

1. Upgraded to HACS 1.17.2
2. Restarted Home Assistant
3. New repositories displayed in HACS UI
4. Browsed and dismissed most, but wanted to download and try "Bom Radar Card"
5. Clicked on "Download" for New Repository
6. Pop-up dialog appeared with "Download" disabled (greyed out)
7. Unable to download card

### Screenshots

![image](https://user-images.githubusercontent.com/39589180/141717951-bc11e24e-b6da-4ef7-9cae-604dcff2fb5c.png)

### Javascript logs from your browser console

```text
card-mod.js?hacstag=1909275243013:1 The main 'lit-element' module entrypoint is deprecated. Please update your imports to use the 'lit' package: 'lit' and 'lit/decorators.ts' or import from 'lit-element/lit-element.ts'. See https://lit.dev/msg/deprecated-import-path for more information.
(anonymous) @ card-mod.js?hacstag=1909275243013:1
card-mod.js?hacstag=1909275243013:6 CARD-MOD 3.0.14 IS INSTALLED 
bom-weather-card.js:2 GENERIC-WEATHER-CARDVersion 0.90a       
layout-card.js?hacstag=156434866231:207 LAYOUT-CARD 2.3.1 IS INSTALLED 
mini-graph-card-bundle.js:1  MINI-GRAPH-CARD  0.10.0
```

**Comment by ludeeus** [comment]
Thare are no repositories in HACS with that name.
What repository is that?

**Comment by peter-ogden** [comment]
Well that is rather odd, seeing as it popped up on HACS? The Repository link on the card goes to https://github.com/Makin-Things/bom-radar-card

**Comment by ludeeus** [comment]
That repository does not exist in HACS <https://hacs.xyz/default_repositories>.
Since it's not working properly for you, please remove it from custom repositories <https://hacs.xyz/docs/faq/custom_repositories>.

**Comment by peter-ogden** [comment]
I seem to have a list of random custom repositories that I never added, including the one that I was interested in. I'll delete them all.

**Comment by ludeeus** [comment]
That is interesting..
It can not be there unless someone/something has added it manually.
Can you list _all_ your custom integrations and custom cards that you use?

**Comment by peter-ogden** [comment]
I managed to delete all the random custom repositories except for the BOM Radar Card! I get the following error when I try deleting:

![image](https://user-images.githubusercontent.com/39589180/141764306-dabfa2b6-c7dd-4a06-acba-40bec989b9f3.png)

**Comment by ludeeus** [comment]
Restart Home Assistant and try again.

**Comment by ludeeus** [comment]
New repositories are considered "custom" until the next restart.
If you don't remove any, and just restart it should clear out most of them.

**Comment by peter-ogden** [comment]
OK, all have now gone except the Bom Radar Card. I've restarted Home Assistant several times, but it is still listed in Custom Repositories. Every attempt to remove it gives the same error "Could not use repository with ID 0 ('NoneType' object has no attribute 'data')"

**Comment by ludeeus** [comment]
Try adding "https://github.com/theOzzieRat/bom-radar-card" and/or "https://github.com/Makin-Things/bom-radar-card" as a new custom repository with "Lovelace" as the category
Maybe that will force it to fetch some data.

**Comment by peter-ogden** [comment]
I tried adding both. It doesn't allow me to add "https://github.com/Main-Things/bom-radar-card" as I get an error that it already exits. I was able to add "https://github.com/theOzzieRat/bom-radar-card", but this also doesn't allow me to download the card. When I restart Home Assistant, the OzzieRat version disappears, but the original version is still there.

I'm going to just ignore it for now. Thank you for trying to assist.
